lib/classes/ConfigSchemaHandler/normalizeAjvErrors.test.js

```diff
@@ -0,0 +1,218 @@
+                        },
+                      },
+                      required: ['http'],
+                      additionalProperties: false,
+                    },
+                  ],
+                },
+              },
+            },
+          },
+          'additionalProperties': false,
+        },
+        additionalProperties: false,
+      },
+    },
+    additionalProperties: false,
+  };
+
+  const resolveNormalizedErrors = config => {
+    const validate = resolveValidate(schema);
+    validate(config);
+    if (!validate.errors) return [];
+    return normalizeAjvErrors(validate.errors, config, schema);
+  };
+
+  let errors;
+  before(() => {
+    errors = resolveNormalizedErrors({
+      foo: 'bar',
+      package: { incclude: ['./folder'] },
+      functions: {
+        'invalid name': {},
+        'foo': {
+          events: [
+            {
+              bar: {},
+            },
+            {
+              http: null,
+              method: 'GET',
+            },
+            {
+              http: {
+                other: 'foo',
+              },
+            },
+            {
+              http: 'gets foo',
+            },
+            {
+              http: {
+                method: 'get',
+              },
+            },
+          ],
+        },
+      },
+    });
+  });
+
+  describe('Reporting', () => {
+    it('should report error for unrecognized root property', () =>
+      expect(
+        errors.some(error => {
+          if (error.dataPath !== '') return false;
+          if (error.keyword !== 'additionalProperties') return false;
+          error.isExpected = true;
+          return true;
+        })
+      ).to.be.true);
+    it('should report error for unrecognized deep level property', () =>
+      expect(
+        errors.some(error => {
+          if (error.dataPath !== '.package') return false;
+          if (error.keyword !== 'additionalProperties') return false;
+          error.isExpected = true;
+          return true;
+        })
+      ).to.be.true);
+    it('should report error for invalid function name', () =>
+      expect(
+        errors.some(error => {
+          if (error.dataPath !== '.functions') return false;
+          if (error.keyword !== 'additionalProperties') return false;
+          error.isExpected = true;
+          return true;
+        })
+      ).to.be.true);
+    it('should report error for unrecognized event', () =>
+      expect(
+        errors.some(error => {
+          if (error.dataPath !== ".functions['foo'].events[0]") return false;
+          if (error.keyword !== 'anyOf') return false;
+          error.isExpected = true;
+          return true;
+        })
+      ).to.be.true);
+    it('should report error for improper indentation when defining event', () =>
+      expect(
+        errors.some(error => {
+          if (error.dataPath !== ".functions['foo'].events[1].http") return false;
+          if (error.keyword !== 'anyOf') return false;
```

Comment:
I think original test case was correct, why it was changed?

Reply:
Yes, I changed dataPath from `.functions['foo'].events[1]` to `.functions['foo'].events[1].http`. I also changed keyword from `additionalProperties ` to `anyOf`.

The config provided by user at `.functions['foo'].events[1].http` is `null`. The schema requires it to be `anyOf`: either string or an object. `null` is not and object or string. 

So I think current is correct. Do you think otherwise?

Reply:
So I changed it because I believe current spect is also correct.

Reply:
I think both test cases are correct. I just chose the one that's easiest to implement. It relies on the rule saying the deepest error is the most relevant one.

Reply:
Yes, as I indicated in other comment, let's keep one that was removed, and definitely keep one you've introduced.

Note that we should not focus just on outcome of YAML config format. Configuration can be made in JS, JSON, even TypeScript.

Reply:
I'll keep the case that was removed

Reply:
Hopefully tomorrow this would be implemented

Reply:
Thank you @preshetin !